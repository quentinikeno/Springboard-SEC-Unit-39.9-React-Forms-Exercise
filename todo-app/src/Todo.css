.Todo {
	margin: 2rem;
	padding: 1rem;
	border: 0.0125rem solid rgba(0, 0, 0, 0.25);
}

.Todo-task-completed {
	text-decoration: line-through;
}

.Todo :is(button, span) {
	margin: 0.5rem;
}
