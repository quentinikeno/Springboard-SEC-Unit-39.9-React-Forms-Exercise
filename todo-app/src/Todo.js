const Todo = ({ id, task, removeTodo }) => {
	const handleClick = () => {
		removeTodo(id);
	};
	return (
		<div>
			{task}
			<button className="Todo-delete-btn" onClick={handleClick}>
				X
			</button>
		</div>
	);
};

export default Todo;
