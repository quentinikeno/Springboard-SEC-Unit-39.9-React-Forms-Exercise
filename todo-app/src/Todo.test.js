import { render } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", () => {
	render(<Todo task="Feed cat." />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<Todo task="Feed cat." />);
	expect(asFragment()).toMatchSnapshot();
});
