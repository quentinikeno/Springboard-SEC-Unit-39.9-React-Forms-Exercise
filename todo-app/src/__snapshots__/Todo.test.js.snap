// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div>
      <span
        class="Todo-task"
      >
        Feed cat.
      </span>
      <button
        class="Todo-edit-btn"
      >
        Edit
      </button>
      <button
        class="Todo-completed-btn"
      >
        Mark as completed
      </button>
      <button
        class="Todo-delete-btn"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
